### PR TITLE
BUG Update RemoteFileFormFactoryTest to rely on translation codes rather than actual translation

### PR DESCRIPTION
**Pull request by maxime-rainville**
This only fixes the unit tests, not the behat failure.

Basically, we were target specific translation. I updated the test to rely on the actual translation code, so future translation change won't break the build.

### CODE REVIEW

File: tests/php/Forms/RemoteFileFormFactoryTest.php [hunk 1]
```diff
@@ -145,10 +146,13 @@ public function providerTestRejectedURLs()
...
      */
-    public function testRejectedURLS($config, $rejectedURL, $rejectedMessage)
+    public function testRejectedURLS($config, $rejectedURL, $rejectedMessageCode)
     {
+        // Suppress the warning about us not provding a default translation for _t
+        i18n::config()->set('missing_default_warning', false);
+
         $this->expectException(InvalidUrlException::class);
-        $this->expectExceptionMessage($rejectedMessage);
+        $this->expectExceptionMessage(_t(RemoteFileFormFactory::class . '.' . $rejectedMessageCode));
```
Comment (ScopeyNZ) [inline tests/php/Forms/RemoteFileFormFactoryTest.php]: Can you leave in the defaults too? Related https://github.com/silverstripe/silverstripe-framework/pull/8949

**Comment by dnsl48** [comment]
are you targeting `1` or `1.4`?

File: tests/php/Forms/RemoteFileFormFactoryTest.php (hunk 1 above)
Comment (dnsl48) [reply tests/php/Forms/RemoteFileFormFactoryTest.php]: yes, I agree, if we could take out of equation the actual translation functionality that would make this test more _unitish_ :)

Comment (dnsl48) [reply tests/php/Forms/RemoteFileFormFactoryTest.php]: though, I don't believe we should leave the default messages in this test. We'll have to refactor RemoteFileFormFactory and put those into the class constants so they are reusable externally. Not sure if that's worth a shot.

Comment (robbieaverill) [reply tests/php/Forms/RemoteFileFormFactoryTest.php]: The proposed change here looks like a good stop gap. Ultimately @ScopeyNZ's proposed change in https://github.com/silverstripe/silverstripe-framework/pull/8949 is what we need to do to isolate translations in tests.

Comment (dnsl48) [reply tests/php/Forms/RemoteFileFormFactoryTest.php]: I don't see any value in having `$this->expectExceptionMessage` assertion in this test, except if tests actual translations. I reckon we should either remove it, or set `$usesTranslations = true;` in this test case.

Comment (ScopeyNZ) [reply tests/php/Forms/RemoteFileFormFactoryTest.php]: The only reason that I ask we maintain the default is that it makes it easier to overload the translation functionality. Having `_t` in tests doesn't feel right, and having translated exception messages also feels strange.

I agree with removing the message assertion. `$usesTranslations` doesn't exist yet so there's no point setting that.

**Comment by unclecheese** [comment]
Brilliant. I love this fix.

**Pull request merged**